src/test/java/TasksServletTest.java

```diff
@@ -112,48 +143,109 @@ public void noTaskLists() throws IOException, ServletException {
     Mockito.when(tasksClient.listTaskLists()).thenReturn(NO_TASKLISTS);
     servlet.doGet(request, response);
     printWriter.flush();
-    List<Task> tasks = gson.fromJson(stringWriter.toString(), LIST_OF_TASKS_TYPE);
-    Assert.assertTrue(tasks.isEmpty());
+    TasksResponse tasksResponse = gson.fromJson(stringWriter.toString(), TasksResponse.class);
+    Assert.assertEquals(NO_TASKLISTS_TITLES, tasksResponse.getTaskListNames());
+    Assert.assertEquals(0, tasksResponse.getTasksToComplete());
+    Assert.assertEquals(0, tasksResponse.getTasksDueToday());
+    Assert.assertEquals(0, tasksResponse.getTasksCompletedToday());
+    Assert.assertEquals(0, tasksResponse.getTasksOverdue());
   }
 
   @Test
-  public void emptyTaskLists() throws IOException, ServletException {
-    Mockito.when(tasksClient.listTaskLists()).thenReturn(SOME_TASKLISTS);
+  public void noTasks() throws IOException, ServletException {
+    Mockito.when(tasksClient.listTaskLists()).thenReturn(ONE_TASKLIST);
     Mockito.when(tasksClient.listTasks(TASKLIST_ONE)).thenReturn(NO_TASKS);
-    Mockito.when(tasksClient.listTasks(TASKLIST_TWO)).thenReturn(NO_TASKS);
     servlet.doGet(request, response);
     printWriter.flush();
-    List<Task> tasks = gson.fromJson(stringWriter.toString(), LIST_OF_TASKS_TYPE);
-    Assert.assertTrue(tasks.isEmpty());
+    TasksResponse tasksResponse = gson.fromJson(stringWriter.toString(), TasksResponse.class);
+    Assert.assertEquals(ONE_TASKLIST_TITLES, tasksResponse.getTaskListNames());
+    Assert.assertEquals(0, tasksResponse.getTasksToComplete());
+    Assert.assertEquals(0, tasksResponse.getTasksDueToday());
+    Assert.assertEquals(0, tasksResponse.getTasksCompletedToday());
+    Assert.assertEquals(0, tasksResponse.getTasksOverdue());
   }
 
   @Test
-  public void oneEmptyTaskList() throws IOException, ServletException {
-    Mockito.when(tasksClient.listTaskLists()).thenReturn(SOME_TASKLISTS);
-    Mockito.when(tasksClient.listTasks(TASKLIST_ONE)).thenReturn(NO_TASKS);
-    Mockito.when(tasksClient.listTasks(TASKLIST_TWO)).thenReturn(TASKS_ONE_TWO);
+  public void taskOverdue() throws IOException, ServletException {
+    Mockito.when(tasksClient.listTaskLists()).thenReturn(ONE_TASKLIST);
+    Mockito.when(tasksClient.listTasks(TASKLIST_ONE)).thenReturn(TASKS_DUE_YESTERDAY);
+    servlet.doGet(request, response);
+    printWriter.flush();
+    TasksResponse tasksResponse = gson.fromJson(stringWriter.toString(), TasksResponse.class);
+    Assert.assertEquals(ONE_TASKLIST_TITLES, tasksResponse.getTaskListNames());
+    Assert.assertEquals(1, tasksResponse.getTasksToComplete());
+    Assert.assertEquals(0, tasksResponse.getTasksDueToday());
+    Assert.assertEquals(0, tasksResponse.getTasksCompletedToday());
+    Assert.assertEquals(1, tasksResponse.getTasksOverdue());
```

Comment:
It's not immediately obvious that these responses are correct, just by looking at this test method. I see that you have incomplete and overdue tasks, but how do I know that `TASKS_DUE_YESTERDAY` only returns one overdue task?

If you follow my suggestion above (to allow comparing responses), then you can define static constants for the expected responses to each set of input tasks. This would put the definition of the responses closer to where the inputs are defined.

(I am not sure what you would do with more complex tests that use multiple task lists - probably just define responses for those combinations.)

Reply:
`RESOLVED` https://github.com/googleinterns/step7-2020/pull/105